internal/services/appservice/app_service_certificate_order_key_vault_store.go

```diff
@@ -0,0 +1,243 @@
+package appservice
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/hashicorp/go-azure-helpers/lang/pointer"
+	"github.com/hashicorp/go-azure-helpers/lang/response"
+	"github.com/hashicorp/go-azure-helpers/resourcemanager/commonids"
+	"github.com/hashicorp/go-azure-helpers/resourcemanager/commonschema"
+	"github.com/hashicorp/go-azure-helpers/resourcemanager/location"
+	"github.com/hashicorp/go-azure-sdk/resource-manager/web/2023-01-01/appservicecertificateorders"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/sdk"
+	keyVaultValidate "github.com/hashicorp/terraform-provider-azurerm/internal/services/keyvault/validate"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/services/web/validate"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/suppress"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/validation"
+)
+
+type CertificateOrderCertificateResource struct{}
+
+type CertificateOrderCertificateModel struct {
+	Name               string `tfschema:"name"`
+	CertificateOrderId string `tfschema:"certificate_order_id"`
+	Location           string `tfschema:"location"`
+	KeyVaultId         string `tfschema:"key_vault_id"`
+	KeyVaultSecretName string `tfschema:"key_vault_secret_name"`
+}
+
+func (r CertificateOrderCertificateResource) Arguments() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{
+		"name": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: validation.StringIsNotEmpty,
+		},
+
+		"certificate_order_id": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: validate.CertificateOrderID,
+		},
+
+		"key_vault_id": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ValidateFunc: commonids.ValidateKeyVaultID,
+			// TODO -- remove when issue https://github.com/Azure/azure-rest-api-specs/issues/28498 is addressed
+			DiffSuppressFunc: suppress.CaseDifference,
+		},
+
+		"key_vault_secret_name": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ValidateFunc: keyVaultValidate.NestedItemName,
+		},
+	}
+}
+
+func (r CertificateOrderCertificateResource) Attributes() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{
+		"location": commonschema.LocationComputed(),
+
+		"type": {
+			Type:     pluginsdk.TypeString,
+			Computed: true,
+		},
```

Comment:
We don't expose the top level `type` field for resources, also this isn't defined in the resource's model struct and isn't being set in the read so this should be removed 
```suggestion
```